src/site/content/en/blog/webassembly-feature-detection/index.md

```diff
@@ -0,0 +1,110 @@
+---
+title: WebAssembly feature detection
+subhead: Learn how to use the newest WebAssembly features while supporting users across all browsers.
+description: Learn how to use the newest WebAssembly features while supporting users across all browsers.
+date: 2022-01-20
+hero: image/9oK23mr86lhFOwKaoYZ4EySNFp02/AQnv9b8SVDwo3y5wSQdG.png
+alt: Spectral lines of phosphorus.
+authors:
+  - rreverser
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - webassembly
+---
+
+WebAssembly 1.0 was released four years ago, but development didn't stop there. New features are added through the [proposal standardization process](https://github.com/WebAssembly/meetings/blob/main/process/phases.md). As is generally the case with new features on the web, their implementation order and timelines can differ significantly between different engines. If you want to use those new features, you need to ensure that none of your users are left out. In this article you’ll learn an approach for achieving this.
+
+Some new features improve code size by adding new instructions for common operations, some add powerful performance primitives, and others improve developer experience and integration with the rest of the web.
+
+You can find the complete list of proposals and their respective stages in the [official repo](https://github.com/WebAssembly/proposals) or track their implementation status in engines on the official [feature roadmap](https://webassembly.org/roadmap/) page.
+
+To ensure that users of all browsers can use your application, you need to figure out which features you want to use. Then, split them up into groups based on browser support. Then, compile your codebase separately for each of those groups. Finally, on the browser side you need to detect supported features and load the corresponding JavaScript and Wasm bundle.
+
+## Picking and grouping features
+
+Let's walk through those steps by picking some arbitrary feature set as an example. Let's say I've identified that I want to use SIMD, threads, and exception handling in my library for size and performance reasons. Their [browser support](https://webassembly.org/roadmap/) is as follows:
+
+<figure>
+{% Img src="image/9oK23mr86lhFOwKaoYZ4EySNFp02/2HAGqxkbSoNhK03itXBK.png", alt="", width="800", height="813" %}
+  <figcaption>
+    Screenshot of the feature table from https://webassembly.org/roadmap/.
```

Comment:
```suggestion
    View this feature table of <a href="https://webassembly.org/roadmap/">webassembly.org/roadmap</a>.
```

Reply:
Should it be "on webassembly.org/roadmap"?

Reply:
Looks like it should be, or at least it was in the Google Docs draft. I thought the "Screenshot of ..." would be better for a title, but either one is fine. I'm going to fixup the suggestion & accept.